azkaban-common/src/main/java/azkaban/container/models/PodTemplateMergeUtils.java

```diff
@@ -42,95 +45,180 @@ private PodTemplateMergeUtils() {
     mergeVolumes(podSpec, podTemplate);
     mergeInitContainers(podSpec, podTemplate);
-    mergeContainerVolumes(podSpec, podTemplate);
-    mergeReadinessProbe(podSpec, podTemplate);
-    mergeLivelinessProbe(podSpec, podTemplate);
-    mergeStartupProbe(podSpec, podTemplate);
+    mergeFlowContainer(podSpec, podTemplate);
   }
 
-  /**
-   * Set the StartupProbe for flowContainer derived from the podTemplate to the podSpec
-   *
-   * @param podSpec     Already created podSpec using the {@link AzKubernetesV1SpecBuilder}
-   * @param podTemplate Instance of the class {@link AzKubernetesV1PodTemplate} to extract items
-   */
-  private static void mergeStartupProbe(V1PodSpec podSpec, AzKubernetesV1PodTemplate podTemplate) {
-    V1Probe containerStartupProbe = podTemplate.getContainerStartupProbe();
-    if (null != containerStartupProbe) {
-      podSpec.getContainers().get(AzKubernetesV1PodTemplate.FLOW_CONTAINER_INDEX)
-          .setStartupProbe(containerStartupProbe);
-    }
+  private static void mergeFlowContainer(V1PodSpec podSpec, AzKubernetesV1PodTemplate podTemplate) {
+    V1Container podSpecFlowContainer = podSpec.getContainers()
+        .get(AzKubernetesV1PodTemplate.FLOW_CONTAINER_INDEX);
+    V1Container podTemplateFlowContainer = podTemplate.getFlowContainer();
+    mergeTemplateAndPodSpecContainer(podTemplateFlowContainer, podSpecFlowContainer);
+    podSpec.setContainers(podTemplate.getAllContainers());
   }
 
   /**
-   * Set the LivelinessProbe for flowContainer derived from the podTemplate to the podSpec
+   * Merge InitContainers from the dynamically generated pod-spec and podTemplate, such that: 1) Add
+   * all the init containers which are only part of podTemplate. 2) Add all the init containers
+   * which are only part of pod-spec. 3) Add the init containers which are part of both pod-spec and
+   * pod-template by merging them such that: a) Skeleton of templateInitContainer is utilized. b)
+   * Environment variables from podSpecInitContainer are added to corresponding
+   * templateInitContainer. c) ImagePullPolicy, Image, and VolumeMounts are overridden from
+   * podSpecInitContainer to templateInitContainer.
    *
    * @param podSpec     Already created podSpec using the {@link AzKubernetesV1SpecBuilder}
    * @param podTemplate Instance of the class {@link AzKubernetesV1PodTemplate} to extract items
    */
-  private static void mergeLivelinessProbe(V1PodSpec podSpec,
+  private static void mergeInitContainers(V1PodSpec podSpec,
       AzKubernetesV1PodTemplate podTemplate) {
-    V1Probe containerLivelinessProbe = podTemplate.getContainerLivelinessProbe();
-    if (null != containerLivelinessProbe) {
-      podSpec.getContainers().get(AzKubernetesV1PodTemplate.FLOW_CONTAINER_INDEX)
-          .setLivenessProbe(containerLivelinessProbe);
+    List<V1Container> podSpecInitContainers = podSpec.getInitContainers();
+    if (null != podSpecInitContainers) {
```

Comment:
if (null == podSpecInitContainers) return;

Keep the rest of code with less indentation.